src/dbnode/storage/series/lookup/entry.go

```diff
@@ -59,6 +59,7 @@ type IndexWriter interface {
 // NB: users are expected to use `NewEntry` to construct these objects.
 type Entry struct {
 	relookupAndIncrementReaderWriterCount func() (index.OnIndexSeries, bool)
+	queryableBlockRetriever               series.QueryableBlockRetriever
```

Comment:
Note @robskillington the way I'm exposing the way to get "block states" which we check against for `RequiresColdFlushForBlockStart`, is by having the `Entry` have a pointer to the `dbShard` which implements this `series.QueryableBlockRetriever` type. Let me know of any thoughts here.